syntax = "proto3";

package protos;

import "protos/storage.proto";

service KeyValueApi {
  rpc Get (GetArgs) returns (GetReply) {}
  rpc Put (PutArgs) returns (PutReply) {}
}

enum Code {
  Ok = 0;
  Error = 1;
}

message Status {
  Code code = 1;
  string info = 2;
}

message PutArgs {
  string key = 1;
  Value value = 2;
}

message PutReply {
  Status status = 1;
}

message GetArgs {
  string key = 2;
}

message GetReply {
  Status status = 1;
  Value value = 2;
}
